src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -96,17 +86,11 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        // BC to be removed in 4.0
-        if (null === $this->filesystem) {
-            $this->filesystem = $this->getContainer()->get('filesystem');
-            $baseDir = $this->getContainer()->getParameter('kernel.project_dir');
-        }
-
         $kernel = $this->getApplication()->getKernel();
         $targetArg = rtrim($input->getArgument('target'), '/');
 
         if (!is_dir($targetArg)) {
-            $targetArg = (isset($baseDir) ? $baseDir : $kernel->getContainer()->getParameter('kernel.project_dir')).'/'.$targetArg;
+            $targetArg = $kernel->getContainer()->getParameter('kernel.project_dir').'/'.$targetArg;
```

Comment:
we reached consensus to use getParameter from the kernel container for cache_dir, project_dir etc. It can be an optional constructor arg.. thats a new feature to me.

https://github.com/symfony/symfony/pull/23624#issuecomment-318813073